Project 4/SPARQL-JV.md

```diff
@@ -0,0 +1,389 @@
+
+```
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+PREFIX homedev: <http://homeandgarden.com/ontology/>
+
+SELECT ?productName ?productModelValue ?inventoryValue ?productMSRP ?tentativeSalePrice
+WHERE {
+  ?product ro:instance_of homedev:outdoorFurniture .
+  ?product ro:has_value homedev:inventoryValue .
+  BIND (?productMSRP * 0.8 AS ?tentativeSalePrice)
+}
+```
+**Problem 6-3.**
+Find all constructed languages present in written works, that are not Star Wars lore.
+- borked: to be filled in
+```
+borked
+```
+
+**Problem 6-4.**
+Find all musicians born in 1960s, who were members of some grunge band that was based in Seattle.
+- borked: to be filled in
+```
+borked
+```
+
+**Problem 6-5.**
+- List all senators or congresspersons who held office in a state that is not the state of their birthplace. Return both birthplace state and state in which they held office.
+- This query calls for a disjunction. Be sensitive to what is being negated.
+```
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+PREFIX ontopol: <https://politicalontology.org/schema/ontopol.ttl>
+
+SELECT * WHERE
+{
+{ SELECT ?person ?birthState ?officeState WHERE {
+      ?person ro:has_role ?SenatorRole ;
+      ?SenatorRole a ontopol:politicalOfficerRole ;
+      ?politicalOfficerRole ontopol:authority_in ?State ;
+      BIND(?State AS ?officeState) .
+      ?person ontopol:has_birthplace ?State ;
+      BIND(?State AS ?birthState) .
+      }
+}
+UNION
+{ SELECT ?person ?birthState ?officeState WHERE {
+      ?person ro:has_role ?congressPersonRole ;
+      ?congressPersonRole a ontopol:politicalOfficerRole ;
+      ?politicalOfficerRole ontopol:authority_in ?State ;
+      BIND(?State AS ?officeState) .
+      ?person ontopol:has_birthplace ?State ;
+      BIND(?State AS ?birthState) .
+      }
+}
+FILTER(?birthState != ?officeState)
+```
+
+**Problem 6-6.**
+- find all cars manufactured outside the USA, have 360 camera sensors, and rated by JD Power & Associates as having the highest rating in its class. Group by body type.
+- borked: to be filled in
+```
+borked
+```
+
+
+**Problem 5-1.**
+- A customer is searching for shelving and wants to filter results in a catalog online, with specific functionality and material type and within a price range.
+- Get the USD price for all shelves that are not two-shelf, that cost less than $150 USD, and are made out of wood rather than wire frame.
+```
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+PREFIX homedev: <http://homeandgarden.com/ontology/>
+
+SELECT ?tallShelf ?salePriceValueUSD
+WHERE {
+  ?shelf ro:instance_of homedev:Shelf ;
+           ro:bearer_of ?function ;
+           ro:composed_primarily_ of ?material ;
+           homedev:has_sale_price ?salePriceValueUSD .
+  FILTER (?function != homedev:twoShelf && 
+          ?salePriceValueUSD < 150 && 
+          ?material = homedev:woodMaterial && 
+          ?material != homedev:metalwireMaterial)
```

Comment:
Food for thought: should we force "composed_primarily_of" to take at most one value?

Reply:
Good question. The relation was directly from the real Relations Ontology. It might be functional, and so the second is redundant?